server/core/include/scoped_permission.hpp

```diff
@@ -33,6 +35,7 @@ namespace irods::experimental
             : comm_{_comm}
             , path_{_path}
             , old_perm_{filesystem::perms::null}
+            , qualified_username_{fmt::format("{}#{}", comm_.clientUser.userName, comm_.clientUser.rodsZone)}
```

Comment:
I was trying to think if there was any potential issues with this, but I don't think there are.

I think we have to assume that within the server, the RsComm ALWAYS contains a proper username and zone for the client.

You agree?

Reply:
No, I think this is fine.

I think we have the tools to test this in main and 4-3-stable now. We can implement MSIs that are called from the test_federation.py script.

See https://github.com/irods/irods/blob/95dac9a16094d33f4a534572e223440370045b76/plugins/microservices/src/test_scoped_client_identity.cpp.

Reply:
Done.

Empty zone names are converted to the local zone for a qualified username every time. Unit test added to confirm behavior.